tests/UriTest.php

```diff
@@ -587,14 +587,16 @@ public function testPathStartingWithTwoSlashesAndNoAuthorityIsInvalid()
         (new Uri)->withPath('//foo');
     }
 
+    /**
+     * @expectedException \InvalidArgumentException
```

Comment:
This does not assert when the error is raised. So we need to keep using the the method. Maybe you could create private helper method that calls the new `expectException` when it exists and otherwise `setExpectedException`

Reply:
Why does it not assert? If we remove it, the test doesn't pass :thinking:

Reply:
It asserts but it does not distinguish at what time the error occurs. So if it throws earlier than expected, it would still pass.

Reply:
Thanks for the explanation. I’ll create the fake method and roolback the annotations changes

Reply:
@carusogabriel would you have time to finish this?

Reply:
@Tobion Can you please finish this patch? I'll be out in vacations from work for almost 2 weeks :)